src/gamma_inc.jl

```diff
@@ -0,0 +1,580 @@
+        return rangereduce ? t*w/a : a*(w+1)
+    else
+        top = @horner(t , .577215664901533E+00 , -.409078193005776E+00 , -.230975380857675E+00 , .597275330452234E-01 , .766968181649490E-02 , -.514889771323592E-02 , .589597428611429E-03)
+        bot = @horner(t , 1.0 , .427569613095214E+00 , .158451672430138E+00 , .261132021441447E-01 , .423244297896961E-02) 
+        w = top/bot
+        return rangereduce ? (t/a)*(w - 1.0) : a*w
+    end
+end
+"""
+    rgammax(a,x)
+
+Evaluation of exp(-x)*x^a/gamma(a) : ``1/\\Gamma(a) e^{-x} x^{a}``
+"""
+function rgammax(a::Float64,x::Float64)
+    if x == 0.0
+        return 0.0
+    elseif a >= 20.0
+        u =x/a
+        if u == 0.0
+            return 0.0
+        end
+        t = (1.0/a)^2
+        t1 = (((0.75*t - 1.0)*t + 3.5)*t - 105.0)/(a*1260.0) #Using Stirling Series : https://dlmf.nist.gov/5.11.1
+        t1 = t1 + a*logmxp1(u)
+        if t1 >= exparg
+            return rt2pin*sqrt(a)*exp(t1)
+        end
+    else
+        t = a*log(x) - x
+        if t < exparg
+            return 0.0
+        elseif a >= 1.0
+            return exp(t)/gamma(a)
+        else
+            return (a*exp(t))*(1.0 + rgamma1pm1(a))
+        end
+    end
+end   
+
+# Reference : 'Computation of the incomplete gamma function ratios and their inverse' by Armido R DiDonato , Alfred H Morris.
+# Published in Journal: ACM Transactions on Mathematical Software (TOMS)
+# Volume 12 Issue 4, Dec. 1986 Pages 377-393
+# doi>10.1145/22721.23109
+
+"""
+    gamma_p(a,x,IND)
+    
+    DLMF: https://dlmf.nist.gov/8.2#E4 , https://dlmf.nist.gov/8.2#E5
+    Wiki: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    IND --> Accuracy desired ; IND=0 means 14 significant digits accuracy , IND=1 means 6 significant digit and IND=2 means only 3 digit accuracy suffices.
+    gamma_p(a,x) or P(a,x) is the Incomplete gamma function ratio given by : ``1/\\Gamma (a) \\int_{0}^{x} e^{-t}t^{a-1} dt``
+"""
+function gamma_p(a::Float64,x::Float64,ind::Integer)
+    iop = ind + 1
+    acc = acc0[iop]
+    if a<0.0 || x<0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a==0.0 && x==0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a*x==0.0 
+        if x<=a
+            return 0.0
+        else
+            return 1.0
+        end               
+    end
+    
+    if a >= 1.0
+        @goto l10
+    elseif a == 0.5
+        @goto l320
+    elseif x < 1.1
+        @goto l110     
+    end
+    r = rgammax(a,x)
+    if r == 0.0
+        return 1.0
+    else
+        @goto l170    
+    end
+
+    @label l10
+     if a >= big1[iop]
+        @goto l20
+     elseif a > x || x >= x0[iop]
+        @goto l30 
+     else
+        twoa = a + a
+        m = trunc(Int,twoa)
+        if twoa != Float64(m)
+            @goto l30 
+        end
+        i = a
+        i = trunc(Int,i)
+        if a == Float64(i)
```

Comment:
```suggestion
        if isinteger(a)
```
and get rid of the lines above.